crates/algebra/src/bijection.rs

```diff
@@ -0,0 +1,94 @@
+//! Bijections between sets of unique elements.
+use std::{
+    collections::{BTreeMap, HashMap},
+    hash::Hash,
+    sync::Arc,
+};
+
+use crate::category;
+
+type Lookup<A, B> = Arc<dyn Fn(&A) -> Option<B>>;
```

Comment:
`Arc` ing here does not make sense. What if the user does not want to pay the "atomic" overhead? Can we switch this a `Box` or any other more generic thing that can be supersed by an `Arc` or any other primitive that the user might bring?

Reply:
The reason for this is to make Bijections easily clonable which they need to be for bijection composition if we want that. What overhead are you talking about, is it more expensive to deref an Arc than a Box?